src/react/components/common/tagsInput/tagsInput.tsx

```diff
@@ -9,30 +9,58 @@ import { ITag } from "../../../../models/applicationState";
 // tslint:disable-next-line:no-var-requires
 const TagColors = require("./tagColors.json");
 
+/**
+ * Interface for model required to work with lower level
+ * tags input component. Rather than name, uses 'id'.
+ * Requires text attribute, which is used to inject
+ * HTML to customize the tags
+ */
 export interface IReactTag {
     id: string;
     text: any;
     color: string;
 }
 
-interface ITagsInputProps {
+/**
+ * Properties required for TagsInput component
+ * tags - ITag[] or stringified ITag[]
+ * onChange - function to call on tags change
+ */
+export interface ITagsInputProps {
     tags: ITag[];
-    onChange: (value: ITag[]) => void;
+    onChange: (tags: ITag[]) => void;
 }
 
-interface ITagsInputState {
+/**
+ * Current state of tags input component
+ * tags - IReactTag[] - tags used in lower level component
+ * currentTagColorIndex - rotates initial color to apply to tags
+ * selectedTag - tag that was most recently double clicked,
+ *     used to populate modal
+ * showModal - boolean to show tag editor modal or not
+ */
+export interface ITagsInputState {
     tags: IReactTag[];
     currentTagColorIndex: number;
     selectedTag: IReactTag;
     showModal: boolean;
 }
 
-const KeyCodes = {
+/**
+ * Key codes used within tag input component
+ * comma - 188 (delimiter for new tag)
+ * enter - 13 (delimiter for new tag)
+ * backspace - 8 (override deletion of tags)
+ */
+export const KeyCodes = {
     comma: 188,
     enter: 13,
     backspace: 8,
 };
 
+/**
+ * Keys that, when pressed, cause creation of new tag
+ */
 const delimiters = [KeyCodes.comma, KeyCodes.enter];
 
 export default class TagsInput extends React.Component<ITagsInputProps, ITagsInputState> {
```

Comment:
Super nit-picky, but I think this reads easier when the following are in the same order that they are when they're defined later on in the code

Reply:
Agreed :) I'll make the fix